flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1074,6 +1129,78 @@ def test_attach_destroyed_volume(self):
         )
         self.assertEqual(exception.args, (volume.blockdevice_id,))
 
+    def test_resize_unknown_volume(self):
+        """
+        ``resize_volume`` raises ``UnknownVolume`` if the supplied
+        ``blockdevice_id`` does not exist.
+        """
+        # blockdevice_id = unicode(uuid4)
+        # exception = self.assertRaises(
+        #     UnknownVolume,
+        #     self.api.resize_volume, blockdevice_id=blockdevice_id
+        # )
+        # self.assertEqual(exception.args, (blockdevice_id,))
+        1/0
+
+    def test_resize_attached_volume(self):
+        """
+        ``resize_volume`` raises ``AlreadyAttached`` if the supplied
+        ``blockdevice_id`` is for a ``BlockDeviceVolume`` that is attached to a
+        host.
+
+        XXX: Should there be a similar test for an attempt to destroy an attached volume?
```

Comment:
Are actual volume providers unable to resize attached volumes?  Or will they succeed and this is merely something we should never ask them to do.  If the latter, we shouldn't make the interface require that they give us an exception.